packages/maskbook/src/plugins/NFT/define.tsx

```diff
@@ -0,0 +1,21 @@
+import { PluginConfig, PluginStage, PluginScope } from '../types'
+import { isTypedMessageAnchor, isTypedMessageText } from '../../protocols/typed-message/types'
+import { pluginName, identifier } from './constants'
+import NFTInPost from './UI/NFTInPost'
+import { getRelevantUrl } from './utils'
+
+export const NFTPluginsDefine: PluginConfig = {
+    pluginName,
+    identifier,
+    stage: PluginStage.Beta,
+    scope: PluginScope.Internal,
+    successDecryptionInspector: function Comp(props) {
+        let nftUrl: URL | null = null
+
+        if (isTypedMessageText(props.message)) nftUrl = getRelevantUrl(props.message.content)
+        else if (isTypedMessageAnchor(props.message)) nftUrl = getRelevantUrl(props.message.href)
```

Comment:
extract to new function.

Reply:
what do you mean?

Reply:
I have created a new function `getTypedMessageContent` in `protocols/typed-message/types` which will return content for any `TypedMessage`.